packages/schema/src/doc.ts

```diff
@@ -246,6 +270,29 @@ export class JupyterGISDoc
     }
   }
 
+  private _layerTreeObserver(events: Y.YEvent<any>[]): void {
+    const changes: Array<{
+      id: string;
+      newValue: IJGISLayerGroup;
+    }> = [];
+    let needEmit = false;
+    events.forEach(event => {
+      event.keys.forEach((change, key) => {
+        if (!needEmit) {
+          needEmit = true;
+        }
+        changes.push({
+          id: key as string,
+          newValue: JSONExt.deepCopy(event.target.toJSON()[key])
+        });
+      });
+    });
+    needEmit = changes.length === 0 ? true : needEmit;
```

Comment:
It doesn't seem `needEmit` is needed, or am I missing something?

Reply:
Right, thanks

Reply:
Right, thanks.